scripts/stations.js

```diff
@@ -133,35 +134,90 @@ async function openStationMenu(stationSerialNumber) {
 		menu.innerHTML = `
             <img src="assets/images/gira_footer.svg" alt="footer" id="graphics">
             <div id="stationName">${stationObj.name}</div>
+			<div id="navigateToButton" onclick="routeToStation('${stationSerialNumber}')"><i class="bi bi-sign-turn-right"></i></div>
             <img id="docksImage" src="assets/images/mGira_station.png" alt="Gira station" width="25%">
             <div id="docksButton">${numDocks === 1 ? "1 doca" : `${numDocks} docas`}</div>
             <img id="bikesImage" src="assets/images/mGira_bike.png" alt="Gira bike" width="25%">
-            <div id="bikesButton${stationObj.docks === 0 ? 'Disabled"' : `" onclick="openBikeList('${stationSerialNumber}')"`} >
+            <div id="bikesButton${
+							stationObj.docks === 0 ? 'Disabled"' : `" onclick="openBikeList('${stationSerialNumber}')"`
+						} >
 				${numBikes === 1 ? "1 bicicleta" : `${numBikes} bicicletas`}
 			</div>`;
 	} else {
 		menu.innerHTML = `
             <div id="availableBikesNumber">Ocorreu um erro.</div>
-            <div id="cancelButton" onclick="document.getElementById('stationMenu').remove()">Voltar</div>
+            <div id="cancelButton" onclick="hideStationMenu()">Voltar</div>
         `.trim();
 	}
 
+	// Set that the user has not clicked navigate to station button
+	userClickedNavigateToStation = false;
+
 	// Add swipe event for hiding the station card
 	addSwipeEvent(
 		menu,
 		() => {
-			menu.classList.add("smooth-slide-to-left");
-			setTimeout(() => menu.remove(), 500); // remove element after animation
-			document.getElementById("zoomControls").classList.add("smooth-slide-down-zoom-controls"); // move zoom controls back down
+			hideStationMenu();
 		},
 		() => {
-			menu.classList.add("smooth-slide-to-right");
-			setTimeout(() => menu.remove(), 500); // remove element after animation
-			document.getElementById("zoomControls").classList.add("smooth-slide-down-zoom-controls"); // move zoom controls back down
+			hideStationMenu(true);
 		}
 	);
 }
 
+function hideStationMenu(exitToRight = false) {
+	const menu = document.getElementById("stationMenu");
+
+	if (menu) {
+		// animate menu exiting
+		if (exitToRight) menu.classList.add("smooth-slide-to-right");
+		else menu.classList.add("smooth-slide-to-left");
+
+		// remove element after animation
+		setTimeout(() => {
+			menu.remove(); // hide html element
+		}, 500);
```

Comment:
sabes que eu não gosto muito de timeouts arbitrários. Fui ver e podes dar listen a um evento que te diz quando a animação para, mas tenta testar isto porque eu nunca usei. Ref: https://developer.mozilla.org/en-US/docs/Web/API/Element/transitionend_event
Tirei isto do GPT, take it with a grain of salt mas segundo a documentação parece certo
```suggestion
		// remove element after animation
		menu.addEventListener('transitionend', function(event) {
    		// Check if the transition property that ended is the one you are interested in
    		if (event.propertyName === 'transform') {
     		   // Remove the menu element after the sliding animation finishes
    		    menu.remove();
		    }
		});
```

Reply:
neste caso não era assim tão arbitrário porque a animação durava exatamente 500ms, mas se há uma melhor forma de fazer, why not